src/ir.cpp

```diff
@@ -11432,6 +11432,17 @@ static TypeTableEntry *ir_analyze_bit_shift(IrAnalyze *ira, IrInstructionBinOp *
     } else {
         TypeTableEntry *shift_amt_type = get_smallest_unsigned_int_type(ira->codegen,
                 op1->value.type->data.integral.bit_count - 1);
+        if (bin_op_instruction->op_id == IrBinOpBitShiftLeftLossy &&
+            op2->value.type->id == TypeTableEntryIdComptimeInt) {
+            if (!bigint_fits_in_bits(&op2->value.data.x_bigint,
+                                     shift_amt_type->data.integral.bit_count,
+                                     op2->value.data.x_bigint.is_negative)) {
+                ir_add_error(ira,
+                             &bin_op_instruction->base,
+                             buf_sprintf("RHS of shift is too large for LHS type"));
```

Comment:
this is still missing some of the useful tidbits of the original error message:

` error: integer value 8 cannot be implicitly casted to type 'u3'`

the fact that the value is 8 is helpful, and that it must be casted to `u3` is helpful.

Reply:
True I'll add that in.

Reply:
I suggest having the original error message, and then adding an error note with your extra text.